util/ranger/checker.go

```diff
@@ -32,6 +32,11 @@ func (c *conditionChecker) check(condition expression.Expression) bool {
 	case *expression.ScalarFunction:
 		return c.checkScalarFunction(x)
 	case *expression.Column:
+		if s, ok := condition.(*expression.Column); ok {
+			if s.RetType.EvalType() == types.ETString {
+				return false
```

Comment:
Then we cannot use index for this kind of filter? would it cause performance regression?

Reply:
Yes.  Maybe it cause performance regression, but It is a correctness question.   In Mysql, the range is empty also. If we use index, it can only filter `null` and `''`.